### PR TITLE
[Merged by Bors] - Enable scale codec for SQL proposal txIDs

### PR DESCRIPTION
**Pull request by evt**
## Motivation

Part of #3271 

## Changes
Introduced EncodeSlice/DecodeSlice codec methods + enabled scale codec for SQL proposal txIDs

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by dshulyak** [comment]
xdr2scale binary

File: hare/hare_test.go [hunk 1]
```diff
@@ -378,7 +394,10 @@ func TestHare_onTick_BeaconFromRefBallot(t *testing.T) {
 		}).Times(1)
 	h.mockBlockGen.EXPECT().GenerateBlock(gomock.Any(), lyrID, gomock.Any()).DoAndReturn(
 		func(_ context.Context, _ types.LayerID, got []*types.Proposal) (*types.Block, error) {
-			require.ElementsMatch(t, pList, got)
+			diff := proposalsDiff(t, pList, got)
```
Comment (dshulyak) [inline hare/hare_test.go]: require.Empty

File: hare/hare_test.go [hunk 2]
```diff
@@ -243,6 +251,11 @@ func TestHare_OutputCollectionLoop(t *testing.T) {
 	h.broker.mu.RUnlock()
 }
 
+func proposalsDiff(t *testing.T, pList, got []*types.Proposal) string {
```
Comment (dshulyak) [inline hare/hare_test.go]: tb testing.TB
and then tb.Helper()

File: hare/hare_test.go [hunk 3]
```diff
@@ -243,6 +251,11 @@ func TestHare_OutputCollectionLoop(t *testing.T) {
...
+	less := func(a, b *types.Proposal) bool { return a.ID().String() < b.ID().String() }
+	return cmp.Diff(pList, got, cmpopts.EquateEmpty(), cmp.AllowUnexported(types.Proposal{}, types.Ballot{}, signing.PublicKey{}), cmpopts.SortSlices(less))
```
Comment (dshulyak) [inline hare/hare_test.go]: require.Empty(tb, ...)

File: hare/hare_test.go (hunk 2 above)
Comment (evt) [reply hare/hare_test.go]: done

File: hare/hare_test.go (hunk 1 above)
Comment (evt) [reply hare/hare_test.go]: done

**Comment by evt** [comment]
> xdr2scale binary

removed, cheers

File: hare/hare_test.go (hunk 3 above)
Comment (evt) [reply hare/hare_test.go]: `proposalsDiff` has been removed after merging develop as not used anywhere.

File: hare/hare_test.go [hunk 4]
```diff
@@ -143,6 +143,7 @@ func TestHare_collectOutputAndGetResult(t *testing.T) {
 
 	pids := []types.ProposalID{types.RandomProposalID(), types.RandomProposalID(), types.RandomProposalID()}
 	set := NewSetFromValues(pids...)
+
```
Comment (dshulyak) [inline hare/hare_test.go]: sorry for nitpicking, but changes in unrelated files should be avoided. it makes git history more confusing

**Review by dshulyak — APPROVED** [review]

File: hare/hare_test.go (hunk 4 above)
Comment (evt) [reply hare/hare_test.go]: sure, fixed

**Comment by evt** [comment]
bors try

**Comment by evt** [comment]
bors merge
